apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/introspection/introspection2sdl.kt

```diff
@@ -5,13 +5,24 @@ import okio.buffer
 import okio.sink
 import java.io.File
 
-fun IntrospectionSchema.toSDL(file: File) {
+/**
+ * Previous versions of this method used to encode defaultValues as Json elements. For an example:
+ * "defaultValue": [0, 1]
+ *
+ * This is wrong. The spec mandates defaultValue to be GraphQL encoded String:
+ * "defaultValue": "[0, 1]"
+ *
+ * For backward compatibility, use [legacyDefaultValues] = true
+ */
+fun IntrospectionSchema.toSDL(file: File, legacyDefaultValues: Boolean) {
   file.sink().buffer().use {
-    toSDL(it)
+    toSDL(it, legacyDefaultValues)
```

Comment:
This is not great but at this point, I'm not sure how many wrongly encoded json are in the wild so I went for the tactical approach of making the least invasive change possible